nav2_costmap_2d/src/layered_costmap.cpp

```diff
@@ -275,9 +275,8 @@ bool LayeredCostmap::isCurrent()
 void LayeredCostmap::setFootprint(const std::vector<geometry_msgs::msg::Point> & footprint_spec)
 {
   footprint_ = footprint_spec;
-  nav2_costmap_2d::calculateMinAndMaxDistances(
-    footprint_spec,
-    inscribed_radius_, circumscribed_radius_);
+  std::tie(inscribed_radius_, circumscribed_radius_) = nav2_costmap_2d::calculateMinAndMaxDistances(
```

Comment:
This is in effect the same as before, its still not protected from a simultaneous write/read. Needs either atomic access/change or a mutex around the set footprint callback

Reply:
Yes, of course you are right. I'll protect the assignments.